### ISSUE TITLE
BUG: Dataframe.apply(func, axis=1) when func returns a Series, apply replaces all the rows in the Dataframe for the result of func(first_row) 

### ISSUE THREAD

**Issue by 0sewa0** [pandas-dev/pandas#35818]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd
df = pd.DataFrame([[4, 9], [1, 2], [3, -1]], columns=['A', 'B'])
def add_columns(series):
    series["C"] = series.A * series.B
    series["D"] = series.A + series.B
    return series
df.apply(add_columns, axis=1)
```

#### Problem description
In pandas '1.1.0' I encountered an issue when using `Dataframe.apply(func, axis=1)` to dynamically insert new columns into a dataframe. After calling `Dataframe.apply(func, axis=1)`, where `func` returns a `Series` (as shown in the Code Sample), every row in the will be a copy of the first row with the new columns added.

So the output of the Sample Code:
   A  B   C   D
0  4  9  36  13
1  4  9  36  13
2  4  9  36  13

#### Expected Output
Each row shouldn't be the same as the first row.
(Code Sample using pandas-1.0.5)
   A  B   C   D
0  4  9  36  13
1  1  2   2   3
2  3 -1  -3   2

#### Output of ``pd.show_versions()``

<details>

>>> pd.show_versions()

INSTALLED VERSIONS
------------------
commit           : d9fff2792bf16178d4e450fe7384244e50635733
python           : 3.8.5.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.7.10-201.fc32.x86_64
Version          : #1 SMP Thu Jul 23 00:58:39 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.1.0
numpy            : 1.19.1
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 19.3.1
setuptools       : 41.6.0
Cython           : None
pytest           : 6.0.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.17.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : 0.8.0
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : 0.4.2
scipy            : 1.5.2
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

**Comment by jreback** [comment]
this is a duplicate issue and patched for 1.1.1

**Comment by simonjayhawkins** [comment]
```
>>> import pandas as pd
>>> pd.__version__
'1.1.1'
>>> df = pd.DataFrame([[4, 9], [1, 2], [3, -1]], columns=['A', 'B'])
>>> def add_columns(series):
...     series["C"] = series.A * series.B
...     series["D"] = series.A + series.B
...     return series
...
>>> df.apply(add_columns, axis=1)
   A  B   C   D
0  4  9  36  13
1  1  2   2   3
2  3 -1  -3   2
>>>
```
